File: 백준/Bronze/28295. 체육은 코딩과목 입니다/체육은 코딩과목 입니다.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? 'dev/stdin' : 'run/input.txt';
const input = fs.readFileSync(filePath).toString().trim().split('\n');

const command_line = ['N', 'E', 'S', 'W'];
let p = 0;

input.map((c) => {
  if (c === '1') {
    p = p === 3 ? 0 : p + 1;
  } else if (c === '2') {
    p = p >= 2 ? p - 2 : p + 2;
  } else if (c === '3') {
    p = p === 0 ? 3 : p - 1;
  }
});

console.log(command_line[p]);

File: 백준/Bronze/28295. 체육은 코딩과목 입니다/체육은 코딩과목 입니다.py
command_line = ['N', 'E', 'S', 'W']
p = 0

for _ in range(10):
    c = int(input())
    if c == 1:
        if p == 3:
            p = 0
        else:
            p += 1
    elif c == 2:
        if p >= 2:
            p -= 2
        else:
            p += 2
    elif c == 3:
        if p == 0:
            p = 3
        else:
            p -= 1

print(command_line[p])